compiler/p/codegen/OMRInstOpCodeProperties.hpp

```diff
@@ -12014,8 +13099,9 @@
    /* .mnemonic    = */ OMR::InstOpCode::plxsd,
    /* .name        = */ "plxsd",
    /* .description =    "Prefixed Load VSX Scalar Doubleword", */
-   /* .opcode      = */ 0x00000000,
-   /* .format      = */ FORMAT_UNKNOWN,
+   /* .prefix      = */ 0x04000000,
+   /* .opcode      = */ 0xA8000000,
+   /* .format      = */ FORMAT_VRT_D34_RA_R,
    /* .minimumALS  = */ TR_Processor::TR_PPCp10,
    /* .properties  = */ PPCOpProp_IsVSX |
```

Comment:
This is not an encoding issue and the existing flags are correct. The new binary encoder does not use properties flags to determine how to encode instructions, so the fact that this writes to a vr is a non-issue in terms of the correctness of this flag. The ISA document makes it clear that this (and the other instructions you pointed out) are specifically part of the VSX instruction set, in that they will not work if VSX support is disabled.

Reply:
to separate VSX from VMX independently is a futile exercise ... let's not linger on the point of disabling VSX. 
as long as it is clear to everybody that the target register has to be a type of TR_VRF, I am fine with it. How to make that clear was my point.

Reply:
do you foresee a potential programmability issue here?  for example, i wanted to do:  fd1 = fd2 + fd3;   (all are double-precision floating point data)
             lfd  fpr2, [fd2]                       <===  TR_FPR
             lxsd or plxsd  vrt3, [fd3]       <===  TR_VRF
             xsadddp  xt1 = xa2 + xb3     <===   TR_VSX_SCALAR
Two problems to me:
1) You will assert somewhere that the type of source registers is incorrect;
2) Encoding the registers correctly; (TR_VRF is the main issue ...  TR_FPR should work ok if no assert)

Reply:
In terms of the asserts, the assertions for VSRs are permissive and will allow registers of type `TR_VRF` and `TR_FPR` to be used without issue [1]. The encoding is also already handled, as the format is the only thing about the instruction the binary encoder will be looking at, and the format tells it that the target register should be encoded as a VR.

The only potential issue I foresee at the moment would be that there's no way to uniformly load a floating point value into a virtual register of type `TR_VSX_SCALAR`. This would be a nice feature to have in situations with high register pressure where using only VRs or FPRs may require a spill and the register is only used in place of a VSR, but I don't think it's particularly urgent. Fixing this would require special handling to be added to the register allocator (probably by turning `lxsd` into `lfd` at RA time if an FPR is selected) and is beyond the scope of this PR, which is focused on getting binary encoding of the prefixed instructions specifically to work.

[1] https://github.com/eclipse/omr/blob/e9c729d0288ddd40b690a9bbf638b46c35f72057/compiler/p/codegen/PPCBinaryEncoding.cpp#L68-L81